core/shared/src/main/scala-2.12/zio/stm/ZSTMUtils.scala

```diff
@@ -0,0 +1,16 @@
+package zio.stm
+
+import scala.collection.mutable
+
+private object ZSTMUtils {
+
+  def newMutableMap[K, V](expectedNumElements: Int): mutable.HashMap[K, V] = {
+    val map = new mutable.HashMap[K, V]
+    if (expectedNumElements > 12) {
```

Comment:
Maybe add a comment with the rationale for those numbers?

Reply:
Actually I did this because I thought Scala 2.12's HashMap initializes the map with a size of 16 like in Java / Scala 2.13+ but it doesn't seem create the underlying arrays at all. I'll change this to always size it